File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

vector<bool> visited;

void dfs(int cur, int n, vector<vector<int>>& computers){
    visited[cur] = true;
    for (int next = 0; next < n; next++){
        if (computers[cur][next] == 1 && !visited[next]){
            dfs(next, n, computers);
        }
    }
}


int solution(int n, vector<vector<int>> computers) {
    visited.assign(n, false);
    int answer = 0;
    for (int i = 0; i < n; i++){
        if (!visited[i]){
            dfs(i, n, computers);
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    visited = [False] * n
    
    def dfs(cur):
        visited[cur] = True
        for nxt in range(n):
            if computers[cur][nxt] == 1 and not visited[nxt]:
                dfs(nxt)
    answer = 0
    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer+=1
    return answer